### DISCUSSION TITLE
Google Play Whisper.TFlite Demo

### DISCUSSION THREAD

**Discussion by StuartIanNaylor** [usefulsensors/openai-whisper#24] [General]
Is there any chance it could also have some options to give inference speed and open files for benchmarking?

**Comment by nyadla-sys** [comment]
I created a demo app with file selection and recording features, but later decided to remove them. If someone with experience in Android app development wants to add these features, it would be relatively easy.

**Reply by StuartIanNaylor** [reply]
Not sure but after looking and thinking really the nnapi prob would be the way to go on Android.

```
import org.tensorflow.lite.Interpreter;
import org.tensorflow.lite.nnapi.NnApiDelegate;

Interpreter.Options options = (new Interpreter.Options());
NnApiDelegate nnApiDelegate = null;
// Initialize interpreter with NNAPI delegate for Android Pie or above
if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
    nnApiDelegate = new NnApiDelegate();
    options.addDelegate(nnApiDelegate);
}

// Initialize TFLite interpreter
try {
    tfLite = new Interpreter(loadModelFile(assetManager, modelFilename), options);
} catch (Exception e) {
    throw new RuntimeException(e);
}

// Run inference
// ...

// Unload delegate
tfLite.close();
if(null != nnApiDelegate) {
    nnApiDelegate.close();
}
```

But yet another framework https://developer.android.com/ndk/guides/neuralnetworks#model
